doc/man3/flux_msg_incref.rst

```diff
@@ -0,0 +1,47 @@
+==========================
+flux_msg_incref(3)
+==========================
+
+.. default-domain:: c
+
+SYNOPSIS
+========
+
+.. code-block:: c
+
+   #include <flux/core.h>
+
+   const flux_msg_t *flux_msg_incref (const flux_msg_t *const_msg)
+
+   void flux_msg_decref (const flux_msg_t *const_msg)
+
+
+DESCRIPTION
+===========
+
+:func:`flux_msg_incref` provides a constant pointer to a `flux_msg_t` object.
+This function prevents `flux_msg_t` objects from being destroyed while response 
+messages (perhaps a `streaming RPC <https://flux-framework.readthedocs.io/
+projects/flux-rfc/en/latest/spec_6.html>`_) are being sent to it.
+
+:func:`flux_msg_decref` allows the message referenced by :func:`flux_msg_incref`
+to be destroyed.
```

Comment:
Cool!  I was just thinking we needed to fill in those missing pages.

Suggestion: add `flux_msg_create()`, `flux_msg_destroy()`, and `flux_msg_copy()` to this page too.  They are all kind of in the constructor/destructor category.

Also I'd keep it simple just say something like
- create creates a message object of the specified type with a reference count of 1.
- incref increments the reference count on a message object
- decref decrements the reference count on a message object and destroy the object if the count reaches 0.
- destroy is an alias for decref

Oh and create should note that type can be FLUX_MSGTYPE_ANY if the type is to be set later with flux_msg_set_type().

To add a page you will also need to update Makefile.am and manpages.py.  Look at another page that contains multiple functions like idset_create(3).  You'll see how the primary page and the stubs are added to both of those files.

Reply:
> decref decrements the reference count on a message object and destroy the object if the count reaches 0.

Thanks so much for this explanation! I was wondering about the difference between decref and destroy. This is quite clever, because it ensures there can be multiple references and the message won't be destroyed unless they're all freed, cool!

Reply:
Actually strike that comment on FLUX_MSGTYPE_ANY.  I'm not sure there's any use for that and it might be something we can clean up.